### ISSUE TITLE
process exit event isn't alway fired. 

### ISSUE THREAD

**Issue by lili21** [nwjs/nw.js#5489]
Neither `win.close(true)` nor `nw.App.quit()`

I need a way to do some clean up work just before the app quit or crashed. 

Here is my code, running in node context ( `bg-script` or `node-main`)
```
process.on('exit', function () {
  // clean up work, 
})

process.on('uncaughtException', function () {
 // clean up work, 
})

```

macOS Sierra
nw.js v0.14.7-sdk

**Comment by divinemanwah** [comment]
Maybe you can try the SIGTERM or SIGKILL events.

**Comment by bastimeyer** [comment]
I ran into this issue today as well when I noticed that my application was not terminating any spawned child processes on application exit and all these spawned processes were being stacked up on each additional app launch.

I'm only able to reproduce this issue when calling `require("nw.gui").App.quit()` though.
When calling `require("nw.gui").Window.get().close(true)` or `process.exit()`, all process exit related listeners are working correctly and a cleanup of all child processes is possible.

Tested with `0.21.6` and `0.25.3` on Linux x64.
I think this bug deserves some attention. Thanks!
